File: ChatGPT/Desarrollo Web Endpoint RESTful/python/endpoint.py
from flask import Flask, request, jsonify
app = Flask(__name__)

@app.route('/saludo', methods=['GET'])
def saludo():
    nombre = request.args.get('nombre', '')
    return jsonify({'mensaje': f'Hola, {nombre}'})

if __name__ == "__main__":
    app.run(debug=True)

# Prueba con:
# curl "http://localhost:5000/saludo?nombre=Juan"

File: ChatGPT/Desarrollo Web Endpoint RESTful/c++/endpoint.cpp
#include "crow.h"
int main() {
    crow::SimpleApp app;
    CROW_ROUTE(app, "/saludo")([](const crow::request& req){
        auto nombre = req.url_params.get("nombre") ? req.url_params.get("nombre") : "";
        return crow::response("{\"mensaje\": \"Hola, " + std::string(nombre) + "\"}");
    });
    app.port(18080).multithreaded().run();
}

// Prueba con:
// curl "http://localhost:18080/saludo?nombre=Juan"
